src/server_manager/install_scripts/install_server.sh

```diff
@@ -249,9 +249,9 @@ function join() {
 function write_config() {
   declare -a config=()
   if [[ $FLAGS_KEYS_PORT != 0 ]]; then
-    config+=("\"portForNewAccessKeys\":$FLAGS_KEYS_PORT")
+    config+=("\"portForNewAccessKeys\": $FLAGS_KEYS_PORT")
   fi
-  config+=$(printf '"hostname": "%q"' ${PUBLIC_HOSTNAME})
```

Comment:
Please revert to using `printf`. That's needed in order to safely escape $PUBLIC_HOSTNAME.

Reply:
Maybe worth putting a comment about it to since I made the same mistake

Reply:
Reverted to using `printf` with the correct syntax for appending to an array. The missing leading parenthesis added the hostname to `config[0]`, preventing `join`from using the provided separator. Also added a comment.

Reply:
![image](https://user-images.githubusercontent.com/93658380/149077067-86ae38c2-9210-49c3-ab5e-a4885d41903b.png)
Hi all, 
I also have the same error, can't fix it, can you help me